cpp/lib/ome/bioformats/out/MinimalTIFFWriter.cpp

```diff
@@ -0,0 +1,308 @@
+        // (2nd generation TIFF, TIFF with 8-byte offsets and big TIFF
+        // respectively).
+        const char *suffixes[] = {"tif", "tiff", "tf2", "tf8", "btf"};
+
+        WriterProperties
+        tiff_properties()
+        {
+          WriterProperties p("MinimalTIFF",
+                             "Baseline Tagged Image File Format");
+
+          p.suffixes = std::vector<boost::filesystem::path>(suffixes,
+                                                            suffixes + boost::size(suffixes));
+
+
+          const PixelType::value_map_type& pv = PixelType::values();
+          std::set<ome::xml::model::enums::PixelType> pixeltypes;
+          for (PixelType::value_map_type::const_iterator i = pv.begin();
+               i != pv.end();
+               ++i)
+            {
+              pixeltypes.insert(i->first);
+            }
+          p.codec_pixel_types.insert(WriterProperties::codec_pixel_type_map::value_type("default", pixeltypes));
+
+          return p;
+        }
+
+        const WriterProperties props(tiff_properties());
+
+      }
+
+      MinimalTIFFWriter::MinimalTIFFWriter():
+        ::ome::bioformats::detail::FormatWriter(props),
+        logger(ome::common::createLogger("MinimalTIFFWriter")),
+        tiff(),
+        ifd(),
+        ifdIndex(0),
+        seriesIFDRange(),
+        bigTIFF(false)
+      {
+      }
+
+      MinimalTIFFWriter::MinimalTIFFWriter(const WriterProperties& writerProperties):
+        ::ome::bioformats::detail::FormatWriter(writerProperties),
+        logger(ome::common::createLogger("MinimalTIFFWriter")),
+        tiff(),
+        ifd(),
+        ifdIndex(0),
+        seriesIFDRange(),
+        bigTIFF(false)
+      {
+      }
+
+      MinimalTIFFWriter::~MinimalTIFFWriter()
+      {
+      }
+
+      void
+      MinimalTIFFWriter::setId(const boost::filesystem::path& id)
+      {
+        FormatWriter::setId(id);
+
+        std::string flags("w");
+
+        // Get expected size of pixel data
+        ome::compat::shared_ptr<const ::ome::xml::meta::MetadataRetrieve> mr(getMetadataRetrieve());
+        storage_size_type pixelSize = significantPixelSize(*mr);
+        // Multiply by 5% to allow for alignment and TIFF metadata overhead.
+        bool needBig = (pixelSize + pixelSize/20) > storage_size_type(std::numeric_limits<uint32_t>::max());
+
+#if TIFF_HAVE_BIGTIFF
+        if (getBigTIFF() || needBig)
+          {
+            flags += "8";
+
+            if (!bigTIFF) // Not set manually
+              boost::format fmt
+                ("Pixel data size is %1%, but TIFF without BigTIFF "
+                 "support enabled has a maximum size of %2%; "
+                 "automatically enabling BigTIFF support to prevent potential failure");
+            fmt % pixelSize % std::numeric_limits<uint32_t>::max();
+
+            BOOST_LOG_SEV(logger, ome::logging::trivial::warning) << fmt.str();
+          }
+        else (!getBigTIFF() && needBig)
```

Comment:
Ah right, sorry I didn't see what you meant there.  Yes, that's definitely wrong.  Should be making use of the fact that bigTIFF is optional, so the first should be "(bigTIFF && *bigTIFF) || (!bigTIFF && needBig)" while the second should be "bigTIFF && !*bigTIFF && needBig".  I'll fix this shortly.